chia/cmds/sim_funcs.py

```diff
@@ -0,0 +1,516 @@
+    elif fingerprint is not None and fingerprint not in fingerprints:
+        print(f"Invalid Fingerprint. Fingerprint {fingerprint} was not found.")
+        return None
+    if auto_generate_key and len(fingerprints) == 1:
+        return fingerprints[0]
+    if len(fingerprints) == 0:
+        if not auto_generate_key:
+            if (
+                input("No keys in keychain. Press 'q' to quit, or press any other key to generate a new key.").lower()
+                == "q"
+            ):
+                return None
+        # generate private key and add to wallet
+        fingerprint = generate_and_return_fingerprint()
+    else:
+        print("Fingerprints:")
+        print(
+            "If you already used one of these keys, select that fingerprint to skip the plotting process."
+            " Otherwise, select any key below."
+        )
+        for i, fp in enumerate(fingerprints):
+            row: str = f"{i + 1}) "
+            row += f"{fp}"
+            print(row)
+        val = None
+        prompt: str = f"Choose a simulator key [1-{len(fingerprints)}] ('q' to quit, or 'g' to generate a new key): "
+        while val is None:
+            val = input(prompt)
+            if val == "q":
+                return None
+            elif val == "g":
+                fingerprint = generate_and_return_fingerprint()
+                break
+            elif not val.isdigit():
+                val = None
+            else:
+                index = int(val) - 1
+                if index < 0 or index >= len(fingerprints):
+                    print("Invalid value")
+                    val = None
+                    continue
+                else:
+                    fingerprint = fingerprints[index]
+        assert fingerprint is not None
+    return fingerprint
+
+
+async def generate_plots(config: Dict[str, Any], root_path: Path, fingerprint: int, bitfield: bool) -> None:
+    """
+    Pre-Generate plots for the new simulator instance.
+    """
+    from time import time
+
+    from chia.simulator.block_tools import BlockTools, test_constants
+    from chia.simulator.start_simulator import PLOT_SIZE, PLOTS
+
+    farming_puzzle_hash = decode_puzzle_hash(config["simulator"]["farming_address"])
+    os.environ["CHIA_ROOT"] = str(root_path)  # change env variable, to make it match what the daemon would set it to
+
+    # create block tools and use local keychain
+    bt = BlockTools(
+        test_constants,
+        root_path,
+        automated_testing=False,
+        plot_dir=config["simulator"].get("plot_directory", "plots"),
+        keychain=Keychain(),
+    )
+    await bt.setup_keys(fingerprint=fingerprint, reward_ph=farming_puzzle_hash)
+    starting_time = time()
+    await bt.setup_plots(
+        num_og_plots=PLOTS, num_pool_plots=0, num_non_keychain_plots=0, plot_size=PLOT_SIZE, bitfield=bitfield
+    )
+    print(f"{'New plots generated.' if time() - starting_time > 5 else 'Using Existing Plots'}\n")

```

Comment:
heh, if I have a machine so fast it can generate test plots in under 5 seconds, it will look like I'm "Using existing plots" :)

Anyway, I see we have a function to generate these plots already. I think the CI should use this as well. Since we pick up existing blocks that may exist already, I believe the `.chia/test_plots` directory is already stored and restored on CI so we wouldn't have to generate them every run

Reply:
fixed